_posts/2022/2022-01-31-infrastructure-software-we-use-to-run-our-saas.md

```diff
@@ -0,0 +1,181 @@
+
+
+### Static Site Hosting: [Netlify](https://netlify.com)
+
+Our static sites are deployed here, we also make some use of a couple of very simple JavaScript “serverless” functions that are called from our applications.
+
+It has a similarly straightforward deployment approach to Heroku, which is one of the main reasons we use it.
+
+
+### CDN: [Fastly](https://fastly.com)
+
+Solid CDN. The main players are all much of a muchness in this arena. We're only (currently) using it for our Rails assets.
+
+
+### Payment Processing: [Stripe](https://stripe.com)
+
+Why would we use anything else to deal with our credit card transactions? Arguably without it, the business would have struggled to get off the ground.
+
+I speak as a man who previously implemented online credit card processing directly with banks. It was horrible.
+
+
+### Code Collaboration: [Github](https://github.com)
+
+Where our code is stored, reviewed and merged. Pretty standard for a modern development team.
+
+We have a couple of Github Actions that we use for linting our code and compressing our images before merging.
+
+We also use the [dependabot](https://github.com/dependabot) functionality Github absorbed.
+
+
+### Image CDN: [Imgix](https://imgix.com)
+
+There’s a bunch of products that perform this task but their API is the most pleasant of any we’ve seen and their pricing is fair. Good and engaged support.
+
+
+### Video CDN: [Mux](https://mux.com)
+
+Our video CDN of choice. Lovely product, good on site player and very nice API.
+
+
+### Video: [Wistia](https://wistia.com)
+
+We serve some of our video marketing assets through here, we also use YouTube.
+
+
+### Cloud Storage: [AWS S3](https://aws.amazon.com)
+
+We use Amazon’s storage because it’s the industry default, we should probably look at alternatives. AWS’s dashboard terrifies me every time I open it.
+
+
+### Screenshots: [Urlbox](https//urlbox.io)
+
+Really decent solution for taking screenshots. Not perfect (we have to do a bunch of special configuration on our end) but a nice API to getting screenshots of websites.
+
+
+### Alerting: [PagerDuty](https://pagerduty.com)
+
+It's _way_ more than we need and the scheduling is complex and byzantine. But reliable and _fine_.
+
+
+### Status: [Statuspage](https://www.atlassian.com/software/statuspage)
+
+I don't much like it, but it does what we need.
+
+
+### Email: [Postmark](https://postmarkapp.com)
+
+Great software from good people, really terrific deliverability for critical, transactional email. Visibly faster than all the other transactional mail services. Also terrific support.
+
+We haven't expanded to use their broadcast email product.
+
+
+### Geolocation: [IPStack](https://ipstack.com)
+
+Simple, cheap, geolocation by IP address.
+
+
+### File uploading: [FileStack](https://filestack.com)
+
+Nice JS-powered uploader, a requirement to avoid using heroku dynos for file upload. We use other (better) solutions for serving the images, files & videos.
```

Comment:
*Heroku

Reply:
*and

Reply:
```suggestion
Nice JS-powered uploader, a requirement to avoid using Heroku dynos for file upload. We use other (better) solutions for serving the images, files and videos.
```